blockstack/blockstack-browser
# Photo Uploader Overhaul

Closes #1608. Closes #1171.

### What This Does

Initially I just planned to attack the two issues above, but this whole component felt really janky to me, so I ended up going down a rabbit hole of improvements. Sorry about that. So in total, this PR does the following:

* Reworks the visuals of the modal to be more clear
* Fixes the issue where uploading a new photo can't be seen until refreshed
* Show photo upload errors to the user
* Show a loader when uploading the photo
* And most importantly, **allows the user to take photos with their webcam**.
    * This is a pretty standard feature on most profile-based sites, and should be especially useful on phones.

And a few misc cleanups:

* Moves photo upload modal to its own component
* Fixes a few cases where we were doing object mutation instead of making new objects
* Remove unused `EditProfilePage` and `ViewProfilePage` components
* Fix some `SimpleButton` propType issues

### Steps to Test

* Attempt to upload a photo, confirm it works
* Attempt to upload a second photo, confirm you immediately see the change
* Attempt to upload a photo too large / while offline, confirm you see an error.
* Attempt to take a photo with your camera, confirm it works.
* Attempt to take a photo with your camera but deny it access, confirm you see an error.
* Attempt to take a photo with a browser that doesn't support it (e.g. IE11), confirm you see an error.

## Screenshots

### New Modal

<img width="461" alt="screen shot 2018-08-21 at 4 06 28 pm" src="https://user-images.githubusercontent.com/649992/44426580-00dfab00-a55d-11e8-8c0a-2bb1fe190ba5.png">

### Upload Error & Loader

![upload-loader-error](https://user-images.githubusercontent.com/649992/44426591-0e953080-a55d-11e8-9171-1129c41fe51b.gif)

### Take a Photo

![webcam-photo](https://user-images.githubusercontent.com/649992/44426608-17860200-a55d-11e8-8b7d-0cfe8eebec27.gif)
